gitmap.go

```diff
@@ -49,18 +49,34 @@ type GitInfo struct {
 	Body            string    `json:"body"`            // The commit message body
 }
 
-// Map creates a GitRepo with a file map from the given repository path and revision.
-// Use blank or HEAD as revision for the currently active revision.
-func Map(repository, revision string) (*GitRepo, error) {
+// Options for the Map function
+type Options struct {
+	Repository        string // Path to the repository to map
+	Revision          string // Use blank or HEAD for the currently active revision
+	GetGitCommandFunc func(args ...string) *exec.Cmd
+}
+
+func NewOptions(repository, revision string) Options {
```

Comment:
Can you 

1. Remove NewOptions
2. Inside Map: Do a `if opts.GetGitCommandFunc == nil then opts.GetGitCommandFunc = the default ...`

Other than that this looks good.

Reply:
Done